service/provider/src/main/java/gov/va/vro/service/provider/MasPollingProcessor.java

```diff
@@ -1,42 +1,152 @@
-      log.info("Collection {} is ready for processing.", payload.getCollectionId());
-      // call pcQueryCollectionAnnots
+    var claimPayload = exchange.getMessage().getBody(MasAutomatedClaimPayload.class);
+
+    boolean isCollectionReady = checkCollectionStatus(claimPayload);
+
+    if (isCollectionReady) {
+      AbdEvidence abdEvidence = getCollectionAnnots(claimPayload);
       // call Lighthouse
       // Combine results and call PDF generation
-      // if a decision is made, call pcOrderExam
+      GeneratePdfResp generatePdfResp = generatePdf(claimPayload, abdEvidence);
+      // call pcOrderExam in the absence of evidence
     } else {
-      log.info("Collection {} is not ready. Requeueing...", payload.getCollectionId());
+      log.info("Collection {} is not ready. Requeue..ing...", claimPayload.getCollectionId());
       // re-request after some time
-      camelEntrance.notifyAutomatedClaim(payload, masDelays.getMasProcessingSubsequentDelay());
+      camelEntrance.notifyAutomatedClaim(claimPayload, masDelays.getMasProcessingSubsequentDelay());
+    }
+  }
+
+  public boolean checkCollectionStatus(MasAutomatedClaimPayload claimPayload) throws MasException {
+
+    log.info("Checking collection status for collection {}.", claimPayload.getCollectionId());
+    boolean isCollectionReady = false;
+    try {
+      List<Integer> collectionIds = new ArrayList<Integer>();
+      collectionIds.add(claimPayload.getCollectionId());
+      var response = masCollectionAnnotsApiService.getMasCollectionStatus(collectionIds);
+      log.info("Collection Status Response : response Size: " + response.size());
+      for (MasCollectionStatus masCollectionStatus : response) {
+        log.info(
+            "Collection Status Response : Collection ID  {} ",
+            masCollectionStatus.getCollectionsId());
+        log.info(
+            "Collection Status Response : Collection Status {} ",
+            masCollectionStatus.getCollectionStatus());
+        if ((MasStatus.PROCESSED)
+            .equals(MasStatus.getMasStatus(masCollectionStatus.getCollectionStatus()))) {
+          isCollectionReady = true;
+        }
+      }
+    } catch (Exception e) {
+      log.error("Error in calling collection Status API ", e);
+      throw new MasException(e.getMessage(), e);
     }
+    return isCollectionReady;
   }
 
-  private boolean checkCollectionStatus(int collectionId) {
-    return new Random().nextBoolean();
+  public AbdEvidence getCollectionAnnots(MasAutomatedClaimPayload claimPayload)
+      throws MasException {
+
+    log.info(
+        "Collection {} is ready for processing, calling collection annotation service ",
+        claimPayload.getCollectionId());
+    AbdEvidence abdEvidence = new AbdEvidence();
+    try {
+      var response =
+          masCollectionAnnotsApiService.getCollectionAnnots(claimPayload.getCollectionId());
+      for (MasCollectionAnnotation masCollectionAnnotation : response) {
+        log.info(
```

Comment:
There are many log statements here. It might bring up security scan problems as some PII info may be in the logs. I suggest adding a TODO note for later log cleanup.

Reply:
Done